### PR TITLE
Allow OnSubmit event handler to return promises

### PR DESCRIPTION
**Pull request by bmamouri**
Simple change to allow async event handlers in type definitions

### CODE REVIEW

**Comment by bluebill1049** [comment]
Thanks for the PR @bmamouri looks like this change break the build (automation).

Screen Shot 2019-08-15 at 6 59 50 pm

**Comment by bluebill1049** [comment]
probably should be void | Promise<void>

File: src/types.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ export type DataType = Record<string, FieldValue>;
 export type OnSubmit<Data extends DataType> = (
   data: Data,
   e: React.SyntheticEvent,
-) => void;
+) => Promise<void>;
```
Comment (stramel) [inline src/types.ts]: `void | Promise<void>`

**Review by stramel — CHANGES_REQUESTED** [review]

**Comment by bluebill1049** [comment]
@stramel those automation tests is pretty useful here 😺

File: src/types.ts (hunk 1 above)
Comment (stramel) [reply src/types.ts]: @bmamouri You should be able to just accept this suggested change now :)

**Comment by bluebill1049** [comment]
@bmamouri can you please accept the change and we can merge it in :)?

**Comment by bmamouri** [comment]
@bluebill1049, Great library mate! Good to see that you are from Sydney too.

@stramel, Thanks for the fix

**Review by bluebill1049 — APPROVED** [review]

**Comment by bluebill1049** [comment]
thanks mate @bmamouri and your contribution ❤️

**Pull request merged**